internal/capability/operator_install.go

```diff
@@ -24,6 +24,18 @@ type operatorData struct {
 	installedNS   []string
 }
 
+type InstallModeTypeAllNamespaces struct {}
+
+type InstallModeTypeOwnNamespace struct {}
+
+type InstallModeTypeSingleNamespace struct {}
+
+type InstallModeTypeMultiNamespace struct {}
+
+type installModeData interface {
+	createFromOpertorData(data operatorData) (operator.OperatorGroupData)
+ }
```

Comment:
would move this to a another file called install_modes.go or something along those lines..